frontend/src/pages/Admin/index.tsx

```diff
@@ -1,47 +1,62 @@
-import React, { FC, useEffect } from "react";
-import { Outlet, useParams } from "react-router-dom";
+import React, { FC, useState, useEffect } from "react";
+import { Outlet, useParams, useOutletContext } from "react-router-dom";
 import apiAxios from "../../utils/apiAxios";
 import { useKeycloak } from "@react-keycloak/web";
 import Sidebar from "../../components/Sidebar";
 import Main from "../../components/Main";
+import Debug from "../../components/Debug";
+import type { KeycloakInstance } from "keycloak-js";
+
+interface OutletContext {
```

Comment:
Nice work. This'll definitely reduce the amount of times we need to call this hook.